File: app/src/main/java/com/example/retailpulseassignment/MainActivity.java
package com.example.retailpulseassignment;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.example.retailpulseassignment.fragments.DownloadFragment;

public class MainActivity extends AppCompatActivity {

    public static final String TAG = MainActivity.class.getSimpleName();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getSupportFragmentManager().beginTransaction()
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
                .replace(R.id.fl_main, new DownloadFragment(), DownloadFragment.TAG)
                .commit();

    }
}

File: app/src/main/java/com/example/retailpulseassignment/views/activities/MainActivity.kt
package com.example.retailpulseassignment.views.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.retailpulseassignment.R
import com.example.retailpulseassignment.views.fragments.DownloadFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //adding base fragment
        supportFragmentManager.beginTransaction()
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
                .replace(R.id.fl_main, DownloadFragment(), DownloadFragment.TAG)
                .commit()

    }
}